easybuild/easyconfigs/v/vcf-report/vcf-report-v1.0.2-Java-11-LTS.eb

```diff
@@ -0,0 +1,33 @@
+name = 'vcf-report'
+version = 'v1.0.2'
+versionsuffix = '-Java-%(javaver)s-LTS'
+
+homepage = 'https://github.com/molgenis/vip-report/'
+description = """Command-line application to generate a report for any VCF (Variant Call Format) file based on a report template"""
+
+toolchain = {'name': 'dummy', 'version': 'dummy'}
+easyblock = 'Binary'
+
+#
+# Example URL:
+# https://github.com/molgenis/vip-report/
+#
+source_urls = [('http://github.com/molgenis/vip-report/releases/download/%s' % (version))]
+sources = [('vcf-report.jar' )]
```

Comment:
Bij eerdere versies over het hoofd gezien, maar dit is gevaarlijk: EasyBuild cacht z'n downloads en er zit geen versie nummer in de bestandsnaam. Bij een volgende v1.0.3 gaat EB hem dan niet opnieuw downloaden, want het bestand is er al. Vervolgens zal de installatie crashen, omdat de checksum niet klopt.
Het alternatief is ofwel het versie nummer in de bestandsnaam bij de release op github ofwel in het *.eb bestand een alternatieve naam opgeven waaronder het bestand gecacht wordt zie: https://easybuild.readthedocs.io/en/latest/Writing_easyconfig_files.html#common-easyconfig-param-sources-alt

Reply:
@pneerincx we doen nu als onderdeel van de release SOP het volgende:
`4. als gedownloade resources dezelfde naam hebben ondanks dezelfde versie, gooi dan de oude weg uit /apps/sources (e.g. rm -rf /apps/sources/v/vcf-decision-tree/vcf-decision-tree.jar)`
dat lijkt mij voldoende?

Reply:
Dat is een overbodige stap als de easyconfig net wat slimmer is. Weggooien maakt ook dat je cache incompleet is. Heeft niet de voorkeur.

Reply:
Verandert met je voorgestelde wijziging het gebruik van de module?

Het gewenste gedrag is om na het laden van de module het volgende te kunnen doen:
`java -jar ${EBROOTVCFMINREPORT}/vcf-report.jar`
en dus niet `java -jar ${EBROOTVCFMINREPORT}/vcf-report-v1.0.2.jar`

Het weggooien van de cache is overigens gebaseerd op onze FAQ https://wiki.gcc.rug.nl/wiki/HPC_deploy#FAQ

Reply:
* Weggooien van de cache staat daar inderdaad beschreven, maar is alleen bedoeld als work around als het bestand bij de bron gewijzigd is zonder een nieuwe release te maken. Dat is lelijk, maar helaas komt het af en toe voor dat een gereleasede versie X ineens niet meer versie X is.
* Een veelvoorkomend pattern is dat de source/bron ingepakt is, gedownload wordt in een bestand met versienummer in de naam en dat daar na uitpakken een binary uit komt zonder versie in de naam, zodat het commando dat je aan moet roepen om iets te starten inderdaad niet gaat veranderen bij een nieuwe versie, want dat is ook niet fijn. Ik weet niet of dat hier goed zou gaan als je "_alternative formats for sources_" zou gebruiken. Zo nee, dan kun je:
  * Ofwel de release aanpassen en stop je de *.jar zonder versie nummer in een *.tar.gz of *.zip o.i.d. met versie nummer in de naam. Vervolgens moet je dan ook het EasyBlock aanpassen naar PackedBinary of BinariesTarball o.i.d.
  * Ofwel `postinstallcmds` gebruiken om een symlink aan de maken `vcf-report.jar` -> `vcf-report-v1.0.2.jar`. Zie bijvoorbeeld de easyconfig voor Sambamba voor een voorbeeld.

Om wille van reproduceerbaarheid heeft het de voorkeur om een complete cache te hebben: als de bron (tijdelijk/permanent) offline gaat of de code aanpast zonder nieuwe release en je wil iets wat je eerder op het ene cluster aan de praat had kunnen reproduceren op een andere site dan kan dat nog als de cache compleet is: kwestie van cache rsyncen naar nieuwe site. Bij een incompleet EasyBuild recept of incomplete cache wordt dat al snel mission impossible.